pkg/ccl/multitenantccl/tenantcostclient/tenant_side.go

```diff
@@ -790,6 +792,13 @@ func (c *tenantSideCostController) OnResponseWait(
 		c.mu.consumption.RU += float64(readRU)
 	}
 
+	// Record the number of RUs consumed by the IO request.
+	if sp := tracing.SpanFromContext(ctx); sp != nil && sp.RecordingType() != tracingpb.RecordingOff {
```

Comment:
This should go above the lock, since we want to release the lock as quickly as possible.